File: user-service/src/main/java/com/example/user/controller/AuthController.java
package com.example.user.controller;

import com.example.common.dto.ApiResponse;
import com.example.user.dto.KakaoLoginRequest;
import com.example.user.dto.KakaoLoginResponse;
import com.example.user.service.AuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login/kakao")
    public ApiResponse<KakaoLoginResponse> kakaoLogin(@Valid @RequestBody KakaoLoginRequest request) {
        KakaoLoginResponse response = authService.kakaoLogin(request.getAuthCode());
        return ApiResponse.success("로그인 성공", response);
    }
}

File: apps/user-service/src/auth/auth.controller.ts
import { Controller, Post, Body } from '@nestjs/common';
import { AuthService } from './auth.service';
import { KakaoLoginRequestDto } from './dto/kakao-login-request.dto';
import { KakaoLoginResponseDto } from './dto/kakao-login-response.dto';
import { ApiResponse } from '@app/common';

@Controller('api/user/auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post('login/kakao')
  async kakaoLogin(
    @Body() request: KakaoLoginRequestDto,
  ): Promise<ApiResponse<KakaoLoginResponseDto>> {
    const response = await this.authService.kakaoLogin(request.authCode);
    return ApiResponse.success('로그인 성공', response);
  }
}
